chromaui/chromatic-e2e
# Cypress: Pass archive along instead of re-bufferizing it

#AP-4729

## Why This Change?

Many folks (see #128) have had Cypress test runs error out when they add Chromatic. This is happening because if Chrome Devtools Protocol fails to send a method, we [add an entry](https://github.com/chromaui/chromatic-e2e/blob/main/packages/shared/src/resource-archive/index.ts#L63) on the archives that has an `error` but no `body`. However, in Cypress we have been [assuming](https://github.com/chromaui/chromatic-e2e/blob/main/packages/cypress/src/index.ts#L46) all archive entries have a body, so when we bufferize the body, things error out.


## What Changed


1. In Cypress, we no longer try to rebufferize the archived resources' `body`-s. This rebufferize code was added when we were [listening for Cypress resource requests client-side](https://github.com/chromaui/chromatic-e2e/pull/29), so after sending to the server we needed to do a `Buffer.from` on them. But since we are now using CDP with Cypress, we [already bufferize](https://github.com/chromaui/chromatic-e2e/blob/main/packages/shared/src/resource-archive/index.ts#L141) the resource `body`.
2. Renamed the `Watcher` [class](https://github.com/chromaui/chromatic-e2e/pull/168/files#diff-54c7ab2fe39fed47a6094a512c8f647e394bab0f8ad7c1711dc5cc11217ddb1dR25) (where we listen for the network and put things on the archive) to `ResourceArchiver`. This was to a) make clear what the class does and b) align this exported class name with the file name for consistency.

## How to test


- [x] Run Chromatic to make sure snapshots haven't regressed

@tevanoff @mea36 I really wanted to put in a test that covers this case (CDP sends error without breaking the entire test run), but I wasn't able to pin down what exactly was causing the CDP method to fail (there's the `error: invalid InterceptionId` that CDP hits, but I could not figure out how to reliably tell when it will happen). I spent 30 minutes trying to nail that down, but couldn't see how to set up a test in such a way that would recreate that issue, so I left off on that.
